tests/pytorch/test_pytorch_autolog.py

```diff
@@ -0,0 +1,160 @@
+import os
+import pytest
+import pytorch_lightning as pl
+import torch
+from iris import IrisClassification
+import mlflow
+import mlflow.pytorch
+from pytorch_lightning.callbacks.early_stopping import EarlyStopping
+from pytorch_lightning.callbacks import ModelCheckpoint
+
+NUM_EPOCHS = 20
+
+
+@pytest.fixture
+def pytorch_model():
+    mlflow.pytorch.autolog()
+    model = IrisClassification()
+    trainer = pl.Trainer(max_epochs=NUM_EPOCHS)
+    trainer.fit(model)
+    optimizer_name = type(trainer.optimizers[0]).__name__.lower() + "_optimizer"
+    client = mlflow.tracking.MlflowClient()
+    run = client.get_run(client.list_run_infos(experiment_id="0")[0].run_id)
+    return trainer, run, optimizer_name
+
+
+def test_pytorch_autolog_logs_default_params(pytorch_model):
+    _, run, optimizer_name = pytorch_model
+    data = run.data
+    assert "learning_rate_" + optimizer_name in data.params
+    assert "epsilon_" + optimizer_name in data.params
+    assert "optimizer_name" in data.params
+    assert "weight_decay_" + optimizer_name in data.params
+    assert "betas_" + optimizer_name in data.params
+
+
+def test_pytorch_autolog_logs_expected_data(pytorch_model):
+    _, run, _ = pytorch_model
+    data = run.data
+
+    # Checking if metrics are logged
+    assert "loss" in data.metrics
+    assert "val_loss" in data.metrics
+
+    # Testing optimizer parameters are logged
+    assert "optimizer_name" in data.params
+    assert data.params["optimizer_name"] == "Adam"
+
+    # Testing model_summary.txt is saved
+    client = mlflow.tracking.MlflowClient()
+    artifacts = client.list_artifacts(run.info.run_id)
+    artifacts = map(lambda x: x.path, artifacts)
+    assert "model_summary.txt" in artifacts
+
+
+# pylint: disable=unused-argument
+def test_pytorch_autolog_persists_manually_created_run():
+    with mlflow.start_run() as manual_run:
+        mlflow.pytorch.autolog()
+        model = IrisClassification()
+        trainer = pl.Trainer(max_epochs=NUM_EPOCHS)
+        trainer.fit(model)
+        assert mlflow.active_run() is not None
+        assert mlflow.active_run().info.run_id == manual_run.info.run_id
+
+
+def test_pytorch_autolog_ends_auto_created_run(pytorch_model):
+    assert mlflow.active_run() is None
+
+
+@pytest.fixture
+def pytorch_model_with_callback(patience):
+    mlflow.pytorch.autolog()
+    model = IrisClassification()
+    early_stopping = EarlyStopping(monitor="val_loss", mode="min", patience=patience, verbose=True)
+
+    checkpoint_callback = ModelCheckpoint(
+        filepath=os.getcwd(), save_top_k=1, verbose=True, monitor="val_loss", mode="min", prefix=""
+    )
```

Comment:
Can we save the checkpoints in a temporary directory?

Reply:
and can we add `lightning_logs` (generated during the tests) to `.gitignore`?

Reply:
@harupy I will create a follow up PR for autolog and example and i will address all the review comments in the follow up PRs.